src/js/angular/clustermanagement/directives/cluster-list.directive.js

```diff
@@ -103,6 +103,7 @@ function ClusterListComponent($translate, $timeout, $repositories, productInfo,
                         })
                         .catch((error) => {
                             handleErrors(error.data, error.status);
+                            $repositories.deleteLocation(endpoint);
```

Comment:
There is something odd here.
You add a location, which might fail, but this would still create a location, that you have to delete.

On line 237, the logic states that a location adding could be successful and still have errors.
This would not trigger the `catch` though.

Reply:
Also using the `deleteLocation` in the `$repositories` service, has some side effects.

If this is a new location, I guess you can safely delete it using the `LocationsRestService` through `RemoteLocationsService` (there is not delete fn yet there)

Reply:
I also wanted to add something about the method on the line 237 because it's out of scope of this change, but still It probably should be fixed
The method is like this:
```
const addNewLocation = (newLocation) => {
                return RemoteLocationsService.addLocationHttp(newLocation)
                    .then((location) => {
                        if (location.error) {
                            handleErrors(location.error);
                            return;
                        }
                        if (location) {
                            return Location.fromJSON(location);
                        }
                    })
                    .catch((error) => {
                        throw error;
                    });
            };
```
the issue is that you throw a synchronous error from a promise chain which might be problematic and cause unhandled promise rejections. You better do something like this instead
```
catch((error) => {
  return Promise.reject(new Error(error));
})
```

Reply:
added

Reply:
removed the checks on line 237, added the promise rejection